### ISSUE TITLE
[flake8-comprehensions] Unnecessary literal within `set` call

### ISSUE THREAD

**Issue by bluetech** [astral-sh/ruff#19267]
### Summary

Currently there is no rule which warns about these:

```py
set({1, 2, 3})
set({x for x in range(10)})
```

The `set()` wrapper can just be removed because the argument is already a set.

I suggest adding a rule `unnecessary-literal-within-set-call`, following the example of [unnecessary-literal-within-dict-call](https://docs.astral.sh/ruff/rules/unnecessary-literal-within-dict-call/).

**Comment by MichaReiser** [comment]
https://github.com/astral-sh/ruff/issues/18331 seems related

**Comment by bluetech** [comment]
> #18331 seems related

Indeed, I was actually starting to write a rule for that issue, but then noticed this bigger "omission" and decided to suggest a fix for it first. Maybe #18331 can be lumped into this rule later, though it doesn't *exactly* fit.
